pkg/mimir/modules.go

```diff
@@ -719,7 +719,16 @@ func (t *Mimir) initQueryFrontendTopicOffsetsReader() (services.Service, error)
 		return nil, err
 	}
 
-	t.QueryFrontendTopicOffsetsReader = ingest.NewTopicOffsetsReader(kafkaClient, t.Cfg.IngestStorage.KafkaConfig.Topic, t.Cfg.IngestStorage.KafkaConfig.LastProducedOffsetPollInterval, t.Registerer, util_log.Logger)
+	// The Kafka partitions may have been pre-provisioned, so the existing partitions in Kafka may be way more
+	// than the actual number of partitions we use. To improve performance, we only look up the actual partitions
```

Comment:
This comment reads a bit mouthful. Can we rephrase it a bit:

```suggestion
	// The Kafka partitions may have been pre-provisioned. There are may be much more existing partitions in Kafka
	// than the actual number we use. To improve performance, we only look up the actual partitions
```